File: Python3/1800. Maximum Ascending Subarray Sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        ans = prefix = nums[0]

        for i, n in enumerate(nums[1:], 1):
            if n > nums[i - 1]:
                prefix += n
            else:
                prefix = n
            ans = max(ans, prefix)

        return ans

File: Java/1800. Maximum Ascending Subarray Sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int prefix = nums[0], ans = prefix;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                prefix += nums[i];
            } else {
                prefix = nums[i];
            }
            ans = Math.max(ans, prefix);
        }

        return ans;
    }
}